lib/diyepw/diyepw/get_tmy_epw_file.py

```diff
@@ -0,0 +1,67 @@
+import tempfile as _tempfile
+import urllib.request as _request
+import re as _re
+import os as _os
+import shutil as _shutil
+from zipfile import ZipFile as _ZipFile
+from urllib.error import URLError
+
+# Buffer for the EPW catalog, which is a large HTML file that we don't want to have to download anew every time
+# a new EPW file is requested
+_catalog_html = None
```

Comment:
should never use a global in a package.  should find a way to scope this to local.

Reply:
Do you have an idea for how to do that? I couldn't come up with anything reasonable. I need a static variable that persists across function calls so that I don't have to repeatedly download the EnergyPlus site's catalog of EPW files. I'd prefer not to persist it to a file just to avoid defining a global, which is the only alternative I was able to come up with.

Reply:
I did a bit more research on this and the options for static variables in Python seems to be to make them global, to make them class properties, to make them properties of the relevant function objects, or to stash them in a default function argument and take advantage of the fact that argument defaults are only evaluated once.

I don't actually see what advantage any of those alternatives has to the current solution. I know the general arguments as to why global variables are considered harmful, and I agree with them, but a data cache is usually held up as a specific counterexample of where global state is acceptable. It doesn't cause confusing hidden behavior or lead to spaghetti code.

I haven't been able to find anything specific to Python packages on why defining a global variable would be specifically problematic in that context, can you help me understand what the issue is?